src/main/java/nextstep/subway/application/SectionService.java

```diff
@@ -0,0 +1,58 @@
+package nextstep.subway.application;
+
+import lombok.RequiredArgsConstructor;
+import nextstep.subway.common.exception.CustomException;
+import nextstep.subway.common.exception.ResponseCode;
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.SectionRepository;
+import nextstep.subway.domain.Sections;
+import nextstep.subway.domain.Station;
+import nextstep.subway.domain.StationRepository;
+import nextstep.subway.ui.dto.section.CreateSectionRequest;
+import nextstep.subway.ui.dto.section.SectionResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
+@RequiredArgsConstructor
+@Service
+public class SectionService {
+    private final SectionRepository sectionRepository;
+    private final StationRepository stationRepository;
+    private final LineRepository lineRepository;
+
+    public SectionResponse createSection(long lineId, CreateSectionRequest request) {
+        Line line = findLine(lineId);
+        Station upStation = findStation(request.getUpStationId());
+        Station downStation = findStation(request.getDownStationId());
+
+        Section section = Section.builder()
+                                 .upStation(upStation)
+                                 .downStation(downStation)
+                                 .distance(request.getDistance()).build();
+        line.addSection(section);
+        sectionRepository.save(section);
+
+        return SectionResponse.of(section);
+    }
+
+    public void deleteSection(final Long lineId, final Long stationId) {
+        Sections sections = findLine(lineId).getSections();
+
+        Section removeSection = sections.removeLastSection(stationId);
+
+        sectionRepository.delete(removeSection);
```

Comment:
Sections 에 `orphanRemoval = true` 옵션을 추가하면 아래와 같이도 가능합니다. 😅 
현재 `구간삭제`의 비즈니스 로직은 조금 어색해보이네요

```suggestion
Line line = findLine(lineId);
line.remove(stationId);
```